prov/gni/include/gnix.h

```diff
@@ -654,6 +658,7 @@ struct gnix_fid_stx {
 	struct fid_stx stx_fid;
 	struct gnix_fid_domain *domain;
 	struct gnix_nic *nic;
+	struct gnix_auth_key *auth_key;
```

Comment:
This might be overstepping it on my end, but it made sense to me at the time because endpoints would have an auth key associated with them, but because a shared TX forces those endpoints to use a shared nic, all endpoints under that shared TX need to be using the same auth key.